great_expectations/data_context/data_context/data_context.py

```diff
@@ -384,16 +385,22 @@ def _check_for_usage_stats_sync(self, project_config: DataContextConfig) -> bool
             A boolean signifying whether or not the current DataContext's config needs
             to be persisted in order to recognize changes made to usage statistics.
         """
-        if project_config.anonymous_usage_statistics.explicit_id is False:
-            return True
-
         project_config_usage_stats: Optional[
             AnonymizedUsageStatisticsConfig
         ] = project_config.anonymous_usage_statistics
         context_config_usage_stats: Optional[
             AnonymizedUsageStatisticsConfig
         ] = self.config.anonymous_usage_statistics
 
+        if (
+            project_config_usage_stats.enabled is False
+            or context_config_usage_stats.enabled is False
+        ):
+            return False
+
+        if project_config_usage_stats.explicit_id is False:
+            return True
```

Comment:
Why isn't this part of the above condition?
```python
if (
        project_config_usage_stats.enabled is False
        or context_config_usage_stats.enabled is False
        or project_config_usage_stats.explicit_id is False
        ):
            return False
```

Reply:
They could all be combined but I separated them out just to make it very explicit what each condition was evaluating. Happy to refactor this if desired.

Reply:
No, I trust your reasoning.
Just wanted to make sure it wasn't a mistake.